From 7ec40808549662d02222a5389deaf34674ff628b Mon Sep 17 00:00:00 2001
From: Zhengchun Li <zhengc@codeaurora.org>
Date: Tue, 9 Oct 2018 22:33:30 +0800
Subject: [PATCH] cryptfs_hw: Replace LOGE with SLOGE

---
 cryptfs_hw.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cryptfs_hw.c b/cryptfs_hw.c
index 5d01651..e8987cb 100755
--- a/cryptfs_hw.c
+++ b/cryptfs_hw.c
@@ -244,7 +244,7 @@ int is_ice_enabled(void)

     ret = read(cmdline_fd, cmdline_buf, sizeof(cmdline_buf));
     if(ret < 0) {
-        LOGE("Error reading the file /proc/cmdline\n");
+        SLOGE("Error reading the file /proc/cmdline\n");
         return -1;
     }
     cmdline_buf[ret - 1] = '\0';
--
1.9.1

