From f4e5f744c3622429835cf551a472a1431a4e97e8 Mon Sep 17 00:00:00 2001
From: Himal Ghimiray <himalg@codeaurora.org>
Date: Wed, 22 Mar 2017 10:19:20 +0530
Subject: [PATCH 1/1] sysvinit logs disabled.

---
 init.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/init.c b/src/init.c
index f4f7193..5de42a0 100644
--- a/src/init.c
+++ b/src/init.c
@@ -811,12 +811,14 @@ void console_stty(void)
  */
 void print(char *s)
 {
+#if 0
	int fd;

	if ((fd = console_open(O_WRONLY|O_NOCTTY|O_NDELAY)) >= 0) {
		write(fd, s, strlen(s));
		close(fd);
	}
+#endif
 }

 /*
@@ -2285,7 +2287,7 @@ void check_init_fifo(void)
	 *	This is a convenient point to also try to
	 *	find the console device or check if it changed.
	 */
-	console_init();
+	//console_init();

	/*
	 *	Process request.
@@ -2605,7 +2607,7 @@ void init_main(void)
   rlim.rlim_cur = rlim.rlim_max;
   setrlimit(RLIMIT_CORE, &rlim);

-  console_init();
+  //console_init();

   if (!reload) {
	int fd;
--
1.9.1
