From 7fdfd2ef8764ddfaeb43e53a756af83d42d8ac8b Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Mon, 27 Jan 2014 03:54:01 -0500
Subject: [PATCH] refpolicy: fix real path for fstools

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
Signed-off-by: Joe MacDonald <joe_macdonald@mentor.com>
---
 policy/modules/system/fstools.fc |   11 +++++++++++
 1 file changed, 11 insertions(+)

--- a/policy/modules/system/fstools.fc
+++ b/policy/modules/system/fstools.fc
@@ -1,6 +1,8 @@
 /sbin/badblocks		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/blkid		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
+/sbin/blkid\.util-linux	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/blockdev		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
+/sbin/blockdev\.util-linux	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/cfdisk		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/dosfsck		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/dump		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
@@ -9,9 +11,11 @@
 /sbin/e4fsck		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/e2label		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/fdisk		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
+/sbin/fdisk\.util-linux	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/findfs		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/fsck.*		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/hdparm		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
+/sbin/hdparm\.hdparm	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/install-mbr	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/jfs_.*		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/losetup.*		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
@@ -24,6 +28,7 @@
 /sbin/mkraid		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/mkreiserfs	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/mkswap		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
+/sbin/mkswap\.util-linux	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/parted		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/partprobe		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/partx		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
@@ -34,6 +39,7 @@
 /sbin/scsi_info		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/sfdisk		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/swapoff		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
+/sbin/swapoff\.util-linux	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/swapon.*		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/tune2fs		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /sbin/zdb		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
@@ -50,7 +56,12 @@
 
 /usr/sbin/clubufflush	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/fatsort	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
+/usr/sbin/findfs		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/parted	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
+/usr/sbin/partprobe		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
+/usr/sbin/partx		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
+/usr/sbin/raw		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
+/usr/sbin/sfdisk		--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 /usr/sbin/smartctl	--	gen_context(system_u:object_r:fsadm_exec_t,s0)
 
 /var/log/fsck(/.*)?		gen_context(system_u:object_r:fsadm_log_t,s0)
