diff -u a/dhcpcd-hooks/Makefile b/dhcpcd-hooks/Makefile
--- a/dhcpcd-hooks/Makefile	2015-09-18 14:47:00.345622850 -0700
+++ b/dhcpcd-hooks/Makefile	2015-09-18 14:47:38.970840854 -0700
@@ -4,7 +4,7 @@

 SCRIPTSDIR=	${LIBEXECDIR}/dhcpcd-hooks
 SCRIPTS=	01-test 02-dump
-SCRIPTS+=	10-mtu 20-resolv.conf 29-lookup-hostname 30-hostname
+SCRIPTS+=	10-mtu 20-resolv.conf 29-lookup-hostname 30-hostname 50-dhcpcd-compat
 SCRIPTS+=	${HOOKSCRIPTS}

 all:

 
--- a/dhcpcd-hooks/50-dhcpcd-compat	2011-01-03 16:24:58.000000000 +0530
+++ b/dhcpcd-hooks/50-dhcpcd-compat	2017-02-08 21:06:13.618273581 +0530
@@ -30,12 +30,12 @@
 esac
 
 if [ "$reason" != "down" ]; then
-	rm -f /var/lib/dhcpcd-"$INTERFACE".info
+	rm -f /data/dhcpcd-"$INTERFACE".info
 	for x in IPADDR INTERFACE NETMASK BROADCAST NETWORK DHCPSID GATEWAYS \
 		DNSSERVERS DNSDOMAIN DNSSEARCH NISDOMAIN NISSERVERS \
 		NTPSERVERS GATEWAY DNS; do
-		eval echo "$x=\'\$$x\'" >> /var/lib/dhcpcd-"$INTERFACE".info
+		eval echo "$x=\'\$$x\'" >> /data/dhcpcd-"$INTERFACE".info
 	done
 fi
 
-set -- /var/lib/dhcpcd-"$INTERFACE".info "$x"
+set -- /data/dhcpcd-"$INTERFACE".info "$x"

--- a/configure	2017-03-10 15:28:27.491601787 +0530
+++ b/configure	2017-03-10 15:30:09.629944443 +0530
@@ -66,9 +66,9 @@
 : ${SBINDIR:=$PREFIX/sbin}
 : ${LIBDIR:=$PREFIX/lib}
 : ${LIBEXECDIR:=$PREFIX/libexec}
-: ${STATEDIR:=/var}
+: ${STATEDIR:=/data}
 : ${DBDIR:=$STATEDIR/db}
-: ${RUNDIR:=$STATEDIR/run}
+: ${RUNDIR:=/var/run}
 : ${MANDIR:=${PREFIX:-/usr}/share/man}
