diff -crB a/nf_nat_rtsp.c b/nf_nat_rtsp.c
*** a/nf_nat_rtsp.c	2012-10-04 17:56:03.946298000 +0530
--- b/nf_nat_rtsp.c	2012-10-04 23:34:38.338519000 +0530
***************
*** 32,37 ****
--- 32,38 ----

  #include <linux/module.h>
  #include <net/tcp.h>
+ #include <net/netfilter/nf_nat.h>
  #include <net/netfilter/nf_nat_helper.h>
  #include <net/netfilter/nf_nat_rule.h>
  #include "nf_conntrack_rtsp.h"
***************
*** 430,436 ****

  static void expected(struct nf_conn* ct, struct nf_conntrack_expect *exp)
  {
!     struct nf_nat_ipv4_multi_range_compat mr;
      u_int32_t newdstip, newsrcip, newip;

      struct nf_conn *master = ct->master;
--- 431,437 ----

  static void expected(struct nf_conn* ct, struct nf_conntrack_expect *exp)
  {
!     struct nf_nat_multi_range_compat mr;
      u_int32_t newdstip, newsrcip, newip;

      struct nf_conn *master = ct->master;
***************
*** 446,455 ****

      mr.rangesize = 1;
      // We don't want to manip the per-protocol, just the IPs.
!     mr.range[0].flags = NF_NAT_RANGE_MAP_IPS;
      mr.range[0].min_ip = mr.range[0].max_ip = newip;

!     nf_nat_setup_info(ct, &mr.range[0], NF_NAT_MANIP_DST);
  }


--- 447,456 ----

      mr.rangesize = 1;
      // We don't want to manip the per-protocol, just the IPs.
!     mr.range[0].flags = IP_NAT_RANGE_MAP_IPS;
      mr.range[0].min_ip = mr.range[0].max_ip = newip;

!     nf_nat_setup_info(ct, &mr.range[0], IP_NAT_MANIP_DST);
  }


