From eea2caa3de02c4191b7ba617ddaba21720716209 Mon Sep 17 00:00:00 2001
From: Divya Theja <divyathe@codeaurora.org>
Date: Mon, 5 Feb 2018 19:30:04 +0530
Subject: [PATCH] Header reference change to base from android-base

---
 fec_verity.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fec_verity.cpp b/fec_verity.cpp
index 3f636dd..7445a39 100644
--- a/fec_verity.cpp
+++ b/fec_verity.cpp
@@ -16,7 +16,7 @@
 
 #include <ctype.h>
 #include <stdlib.h>
-#include <android-base/strings.h>
+#include <base/strings.h>
 #include "fec_private.h"
 
 /* converts a hex nibble into an int */
-- 
1.9.1

