From 56ec3e527f2a03d217d5f07ebb708e6e26fa26ff Mon Sep 17 00:00:00 2001
From: Xin Ouyang <Xin.Ouyang@windriver.com>
Date: Tue, 9 Jun 2015 21:22:52 +0530
Subject: [PATCH] refpolicy: fix real path for sysnetwork

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Xin Ouyang <Xin.Ouyang@windriver.com>
Signed-off-by: Shrikant Bobade <Shrikant_Bobade@mentor.com>
Signed-off-by: Joe MacDonald <joe_macdonald@mentor.com>
---
 policy/modules/system/sysnetwork.fc |    3 +++
 1 file changed, 3 insertions(+)

--- a/policy/modules/system/sysnetwork.fc
+++ b/policy/modules/system/sysnetwork.fc
@@ -54,17 +54,20 @@ ifdef(`distro_redhat',`
 /usr/sbin/dhcdbd		--	gen_context(system_u:object_r:dhcpc_exec_t,s0)
 /usr/sbin/dhcp6c		--	gen_context(system_u:object_r:dhcpc_exec_t,s0)
 /usr/sbin/dhcpcd		--	gen_context(system_u:object_r:dhcpc_exec_t,s0)
 /usr/sbin/ethtool		--	gen_context(system_u:object_r:ifconfig_exec_t,s0)
 /usr/sbin/ifconfig		--	gen_context(system_u:object_r:ifconfig_exec_t,s0)
+/usr/sbin/ifconfig\.net-tools  --  gen_context(system_u:object_r:ifconfig_exec_t,s0)
+/usr/sbin/ip\.iproute2 --  gen_context(system_u:object_r:ifconfig_exec_t,s0)
 /usr/sbin/ip			--	gen_context(system_u:object_r:ifconfig_exec_t,s0)
 /usr/sbin/ipx_configure		--	gen_context(system_u:object_r:ifconfig_exec_t,s0)
 /usr/sbin/ipx_interface		--	gen_context(system_u:object_r:ifconfig_exec_t,s0)
 /usr/sbin/ipx_internal_net	--	gen_context(system_u:object_r:ifconfig_exec_t,s0)
 /usr/sbin/iw			--	gen_context(system_u:object_r:ifconfig_exec_t,s0)
 /usr/sbin/iwconfig		--	gen_context(system_u:object_r:ifconfig_exec_t,s0)
 /usr/sbin/mii-tool		--	gen_context(system_u:object_r:ifconfig_exec_t,s0)
+/usr/sbin/mii-tool\.net-tools  --  gen_context(system_u:object_r:ifconfig_exec_t,s0)
 /usr/sbin/pump			--	gen_context(system_u:object_r:dhcpc_exec_t,s0)
 /usr/sbin/tc			--	gen_context(system_u:object_r:ifconfig_exec_t,s0)
 
 #
 # /var
