diff --git a/sample-etc_ddclient.conf b/sample-etc_ddclient.conf
index 9ba474e..49ecc5e 100644
--- a/sample-etc_ddclient.conf
+++ b/sample-etc_ddclient.conf
@@ -19,12 +19,13 @@
 ##
 ######################################################################
 daemon=300				# check every 300 seconds
-syslog=yes				# log update msgs to syslog
-mail=root				# mail all msgs to root
-mail-failure=root			# mail failed update msgs to root
-pid=/var/run/ddclient.pid		# record PID in file.
-ssl=yes					# use ssl-support.  Works with
-					# ssl-library
+
+#syslog=yes				# log update msgs to syslog
+#mail=root				# mail all msgs to root
+#mail-failure=root			# mail failed update msgs to root
+pid=/var/run/ddclient.pid		# record PID in file.
+#ssl=yes					# use ssl-support.  Works with
+cache=/var/run/ddclient.cache					# ssl-library
 # postscript=script			# run script after updating.  The
 					# new IP is added as argument.
 #
@@ -58,6 +59,8 @@ ssl=yes					# use ssl-support.  Works with
 #
 #use=ip,                     ip=127.0.0.1	# via static IP's
 #use=if,                     if=eth0		# via interfaces
+use=if, if=rmnet_data0
+usev6=if, if=rmnet_data0
 #use=web					# via web
 #
 #protocol=dyndns2				# default protocol
@@ -66,7 +69,9 @@ ssl=yes					# use ssl-support.  Works with
 #server=members.dyndns.org:8245			# default server (bypassing proxies)
 
 #login=your-login				# default login
+login=##Login##
 #password=test					# default password
+password=##Password##
 #mx=mx.for.your.host				# default MX
 #backupmx=yes|no				# host is primary MX?
 #wildcard=yes|no				# add wildcard CNAME?
@@ -76,10 +81,9 @@ ssl=yes					# use ssl-support.  Works with
 ##
 ## (supports variables: wildcard,mx,backupmx)
 ##
-# server=members.dyndns.org,		\
-# protocol=dyndns2			\
+protocol=dyndns2	
+server=members.dyndns.org
 # your-dynamic-host.dyndns.org
-
 ##
 ## dyndns.org static addresses
 ##
