--- a/linux/minidlna.init.d.script.tmpl	2016-10-20 19:19:07.366766200 +0530
+++ b/linux/minidlna.init.d.script.tmpl	2016-10-20 19:55:12.339200304 +0530
@@ -20,7 +20,7 @@

 MINIDLNA=/usr/sbin/minidlnad
 PIDFILE=/var/run/minidlna/minidlna.pid
-CONF=/etc/minidlna.conf
+CONF=/etc/data/minidlna.conf
 ARGS="-f $CONF -R"
 test -f $MINIDLNA || exit 0

--- a/minidlna.conf	2014-06-06 03:44:30.000000000 +0530
+++ b/minidlna.conf	2017-04-03 11:43:43.075048115 +0530
@@ -25,10 +25,10 @@ media_dir=/opt
 #friendly_name=My DLNA Server

 # set this if you would like to specify the directory where you want MiniDLNA to store its database and album art cache
-#db_dir=/var/cache/minidlna
+db_dir=/var/log/minidlna/cache

 # set this if you would like to specify the directory where you want MiniDLNA to store its log file
-#log_dir=/var/log
+log_dir=/var/log/minidlna

 # set this to change the verbosity of the information that is logged
 # each section can use a different level: off, fatal, error, warn, info, or debug
