From 29797e7df947173323cc15c592766421d565dc68 Mon Sep 17 00:00:00 2001
From: Manish Dewangan <manish@codeaurora.org>
Date: Mon, 17 Oct 2016 14:28:05 +0530
Subject: [PATCH] new:wq

---
 git/include/tinyalsa/pcm.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/git/include/tinyalsa/pcm.h b/git/include/tinyalsa/pcm.h
index 5fcb7d8..58a7f56 100644
--- a/git/include/tinyalsa/pcm.h
+++ b/git/include/tinyalsa/pcm.h
@@ -125,6 +125,7 @@ extern "C" {
  * @ingroup libtinyalsa-pcm
  */
 enum pcm_format {
+    PCM_FORMAT_INVALID = -1,
     /** Signed, 8-bit */
     PCM_FORMAT_S8 = 1,
     /** Signed 16-bit, little endian */
-- 
1.9.1

