diff --git a/meson.build b/meson.build
index 95a1166..1ff632b 100644
--- a/meson.build
+++ b/meson.build
@@ -2193,8 +2193,8 @@ endif
 meson_check_api_docs_sh = find_program('tools/meson-check-api-docs.sh')
 run_target(
         'check-api-docs',
-        depends : [man, libsystemd, libudev],
-        command : [meson_check_api_docs_sh, libsystemd.full_path(), libudev.full_path()])
+        depends : [man, libsystemdq],
+        command : [meson_check_api_docs_sh, libsystemdq.full_path(), ])
 
 ############################################################
 
