Subject: [PATCH] refpolicy: fix real path for netutils

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Xin Ouyang <Xin.Ouyang@windriver.com>
Signed-off-by: Joe MacDonald <joe_macdonald@mentor.com>
---
 policy/modules/admin/netutils.fc | 1 +
 1 file changed, 1 insertion(+)

--- a/policy/modules/admin/netutils.fc
+++ b/policy/modules/admin/netutils.fc
@@ -1,10 +1,11 @@
 /bin/ping.* 		--	gen_context(system_u:object_r:ping_exec_t,s0)
 /bin/tracepath.*		--	gen_context(system_u:object_r:traceroute_exec_t,s0)
 /bin/traceroute.*	--	gen_context(system_u:object_r:traceroute_exec_t,s0)
 
 /sbin/arping		--	gen_context(system_u:object_r:netutils_exec_t,s0)
+/bin/arping		--	gen_context(system_u:object_r:netutils_exec_t,s0)
 
 /usr/bin/arping		--	gen_context(system_u:object_r:netutils_exec_t,s0)
 /usr/bin/lft		--	gen_context(system_u:object_r:traceroute_exec_t,s0)
 /usr/bin/nmap		--	gen_context(system_u:object_r:traceroute_exec_t,s0)
 /usr/bin/ping.* 	--	gen_context(system_u:object_r:ping_exec_t,s0)
