--- linux-3.11/include/linux/usb/usbnet.h	2013-09-02 13:46:10.000000000 -0700
+++ linux-3.11.10-301.fc20.x86_64/include/linux/usb/usbnet.h	2016-01-08 12:54:16.297086256 -0800
@@ -158,6 +158,14 @@
 	int		out;		/* tx endpoint */
 
 	unsigned long	data;		/* Misc driver specific data */
+
+    int (*rx_submit_fixup)(struct usbnet *dev, struct urb *urb, gfp_t flags);
+
+    int (*rx_submit_fixup_avb)(struct usbnet *dev, struct urb *urb, gfp_t flags);
+
+    int (*rx_submit_fixup_av_ctrl)(struct usbnet *dev, struct urb *urb, gfp_t flags);
+
+//    void (*rx_complete_fixup)(struct urb *urb);
 };
 
 /* Minidrivers are just drivers using the "usbnet" core as a powerful
