diff -u a/makedepend.SH b/makedepend.SH
--- a/makedepend.SH	2014-07-16 13:29:26.224361999 -0700
+++ b/makedepend.SH	2014-07-16 13:30:05.456191012 -0700
@@ -172,7 +172,7 @@
 	    -e '/^#.*"\/.*\/"/d' \
 	    -e '/: file path prefix .* never used$/d' \
 	    -e '/^#.*git_version\.h/d' \
-	    -e 's#\.[0-9][0-9]*\.c#'"$file.c#" \
+	    -e 's#\.[0-9][0-9]*\.c$#'"$file.c#" \
 	    -e 's/^[	 ]*#[	 ]*line/#/' \
 	    -e '/^# *[0-9][0-9]* *[".\/]/!d' \
 	    -e 's/^.*"\(.*\)".*$/'$filebase'\$(OBJ_EXT): \1/' \
