From b4a1752dc40983f8e815ccf574d0c3b6dc12d061 Mon Sep 17 00:00:00 2001
From: Tyler Wear <twear@codeaurora.org>
Date: Tue, 26 Aug 2014 10:40:44 -0700
Subject: [PATCH 1/1] M-DMS

---
 upnpdescgen.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/upnpdescgen.c b/upnpdescgen.c
index c00cee0..b6d85f0 100755
--- a/upnpdescgen.c
+++ b/upnpdescgen.c
@@ -147,7 +147,7 @@ static const struct XMLElt rootDesc[] =
 	{"/modelURL", ROOTDEV_MODELURL},
 	{"/serialNumber", serialnumber},
 	{"/UDN", uuidvalue},	/* required */
-	{"/dlna:X_DLNADOC xmlns:dlna=\"urn:schemas-dlna-org:device-1-0\"", "DMS-1.50"},
+	{"/dlna:X_DLNADOC xmlns:dlna=\"urn:schemas-dlna-org:device-1-0\"", "M-DMS-1.50"},
 	{"/presentationURL", presentationurl},	/* recommended */
 	{"iconList", INITHELPER((19+PNPX),4)},
 	{"serviceList", INITHELPER((43+PNPX),3)},
-- 
1.8.2.1

