From e0c3cb82b251cea3995b50f5cedc655ab33e062a Mon Sep 17 00:00:00 2001
From: Tyler Wear <twear@codeaurora.org>
Date: Fri, 31 Oct 2014 13:00:59 -0700
Subject: [PATCH 1/1] initial-notify

---
 minidlna.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
 mode change 100644 => 100755 minidlna.c

diff --git a/minidlna.c b/minidlna.c
old mode 100644
new mode 100755
index c8eed5b..b0ea9b0
--- a/minidlna.c
+++ b/minidlna.c
@@ -1072,7 +1072,7 @@ main(int argc, char **argv)
 #endif
 
 	reload_ifaces(0);
-	lastnotifytime.tv_sec = time(NULL) + runtime_vars.notify_interval;
+	lastnotifytime.tv_sec = time(NULL);
 
 	/* main loop */
 	while (!quitting)
-- 
1.8.2.1

