diff -Naur conntrack-tools-1.0.1-orig/src/conntrack.c conntrack-tools-1.0.1/src/conntrack.c
--- conntrack-tools-1.0.1-orig/src/conntrack.c	2015-07-18 20:40:22.039785504 +0530
+++ conntrack-tools-1.0.1/src/conntrack.c	2015-07-18 20:41:04.019460367 +0530
@@ -856,11 +856,10 @@
 {
 	if (inet_aton(cp, &parse->addr))
 		return AF_INET;
-#ifdef HAVE_INET_PTON_IPV6
 	else if (inet_pton(AF_INET6, cp, &parse->addr6) > 0)
 		return AF_INET6;
-#endif
-	return AF_UNSPEC;
+	else
+		return AF_UNSPEC;
 }
 
 union ct_address {
