diff -Naur conntrack-tools-1.0.1-orig/src/parse.c conntrack-tools-1.0.1/src/parse.c
--- conntrack-tools-1.0.1-orig/src/parse.c	2012-01-04 11:48:15.490258748 -0500
+++ conntrack-tools-1.0.1/src/parse.c	2012-05-07 22:49:51.590626617 -0400
@@ -16,6 +16,8 @@
  * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#include <stddef.h>
+
 #include "network.h"
 
 #include <libnetfilter_conntrack/libnetfilter_conntrack.h>
