*** a/defs.h	Thu Aug 22 13:21:59 2013
--- b/defs.h	Tue Aug 27 21:39:49 2013
***************
*** 222,229 ****
  /* Used by DVMRP */
  #define DEFAULT_METRIC		1	/* default subnet/tunnel metric     */
  #define DEFAULT_THRESHOLD	1	/* default subnet/tunnel threshold  */
! 
! /* Used if no relaible unicast routing information available */
  #define UCAST_DEFAULT_SOURCE_METRIC     1024
  #define UCAST_DEFAULT_SOURCE_PREFERENCE 1024
  
--- 222,232 ----
  /* Used by DVMRP */
  #define DEFAULT_METRIC		1	/* default subnet/tunnel metric     */
  #define DEFAULT_THRESHOLD	1	/* default subnet/tunnel threshold  */
! 
! /* Used if no rmnet subnet mask len available set as 255.255.255.248  */
! #define UCAST_DEFAULT_RMNET_MASK     (u_int32)0xf8ffffff
! 
! /* Used if no relaible unicast routing information available */
  #define UCAST_DEFAULT_SOURCE_METRIC     1024
  #define UCAST_DEFAULT_SOURCE_PREFERENCE 1024
  
