diff --git a/Makefile b/Makefile
index 312dc30..73f28f9 100644
--- a/Makefile
+++ b/Makefile
@@ -52,7 +52,7 @@ include snmp.mk
 
 ## Common
 CFLAGS       += $(MCAST_INCLUDE) $(SNMPDEF) $(RSRRDEF) $(INCLUDES) $(DEFS) $(USERCOMPILE)
-CFLAGS       += -O2 -W -Wall -Werror -fno-strict-aliasing
+CFLAGS       += -O2 -W -Wall -Werror -Wno-deprecated-declarations -fno-strict-aliasing
 #CFLAGS       += -O -g
 LDLIBS        = $(SNMPLIBDIR) $(SNMPLIBS) $(EXTRA_LIBS)
 OBJS          = $(IGMP_OBJS) $(ROUTER_OBJS) $(PIM_OBJS) $(DVMRP_OBJS) \
